examples/impute/plot_iterative_imputer_variants_comparison.py

```diff
@@ -97,19 +95,38 @@
 # with different estimators
 estimators = [
     BayesianRidge(),
-    DecisionTreeRegressor(max_features="sqrt", random_state=0),
-    ExtraTreesRegressor(n_estimators=10, random_state=0),
+    RandomForestRegressor(
```

Comment:
Why adding a `RandomForestRegressor`. It will be super slow. It was one of the reasons to use the `ExtraTreesRegressor` instead.

Reply:
Just to replace DecisionTreeRegressor with a regressor that converges. DecisionTreeRegressor was not converging even with tolerance of 0.5 and 250 iterations.

Reply:
So we can replace the `DecisionTreeRegressor` by a non-linear regressor. Basically using a `RandomForestClassifier` would be equivalent to MissForest that was mentioned to be too slow in the docstring (and replace by `ExtraTreesRegressor`). Let me check which other learner could we use there.

Reply:
I tested IterativeImputer implementation with  np.percentile(np.abs(X[~mask_missing_values])), 99) instead of np.max(np.abs(X[~mask_missing_values])). Even DecisionTreeRegressor is converging.

Reply:
These are current runtimes:
BayesianRidge :5.6260459423065186
RF: 2.5009994506835938
ET: 6.401785135269165
KNN: 1.7558178901672363
I thought RF will be the slowest but for current data it performs faster than ET.

Reply:
I don't know what are you referring to with `np.percentile(...)`.

As I say, we only want `RandomForestClassifier` or `ExtraTreesClassifier` but not both.
A non-linear classifier that works in the original setting is:

```python
make_pipeline(Nystroem(kernel="polynomial", degree=2, random_state=0), Ridge(alpha=1e3)),
```

Reply:
In any case, we need to change the discussion on the top of the example to reflect the changes that we are doing here.